astropy_helpers/openmp_helpers.py

```diff
@@ -39,69 +41,222 @@
 }
 """
 
+def _get_flag_value_from_var(flag, var, delim=' '):
+    """
+    Utility to extract `flag` value from `os.environ[`var`]` or, if not present,
+    from `distutils.sysconfig.get_config_var(`var`)`.
+    E.g. to get include path: _get_flag_value_from_var('-I', 'CFLAGS')
+    might return "/usr/local/include".
 
-def add_openmp_flags_if_available(extension):
+    Notes
+    -----
+    Not yet tested and therefore not yet supported on Windows
     """
-    Add OpenMP compilation flags, if available (if not a warning will be
-    printed to the console and no flags will be added)
 
-    Returns `True` if the flags were added, `False` otherwise.
+    if sys.platform.startswith('win'):
+        return None
+
+    # Simple input validation
+    if not var or not flag:
+        return None
+    l = len(flag)
+    if not l:
+        return None
+
+    # Look for var in os.eviron
+    try:
+        flags = os.environ[var]
+    except KeyError:
+        flags = None
+    
+    # If os.environ was unsuccesful try in sysconfig
+    if not flags:
+        try:
+            flags = get_config_var(var)
+        except KeyError:
+            return None
+
+    # Extract flag from {var:value}
+    if flags:
+        for item in flags.split(delim):
+            if flag in item[:l]:
+                return item[l:]
+
+    return None
+
+def _get_include_path():
+    return _get_flag_value_from_var('-I', 'CFLAGS')
+
+def _get_library_path():
+    return _get_flag_value_from_var('-L', 'LDFLAGS')
+
+def get_openmp_flags():
     """
+    Utility for returning compiler and linker flags possibly needed for
+    OpenMP support.
 
-    ccompiler = new_compiler()
-    customize_compiler(ccompiler)
+    Returns
+    -------
+    result : `{'compiler_flags':<flags>, 'linker_flags':<flags>}`
 
-    tmp_dir = tempfile.mkdtemp()
+    Notes
+    -----
+    The flags returned are not tested for validity, use
+    `test_openmp_support(openmp_flags=get_openmp_flags())` to do so.
+    """
 
-    start_dir = os.path.abspath('.')
+    compile_flags = []
+    include_path = _get_include_path()
+    if include_path:
```

Comment:
Could you add a comment explaining why you need to add library and include paths - e.g. does Python ignore environment variables by default? Sorry for being picky with comments, but we've been burned in the past with undocumented workarounds for issues in astropy-helpers, so I'm just making sure this is understandable by other developers :)

Reply:
I don't think commenting in `get_openmp_flags` is correct, semantically the use for why you want these flags belongs to the caller as they don't actually have to be passed to a compiler at this stage, they may just be checked or printed etc. `_get_flag_value_from_var` already docs extraction method, i.e. from `os.environ[`var`]` or, if not present, from `distutils.sysconfig.get_config_var(`var`)`.

I believe the level of doc you want already exists [here](https://github.com/jamienoss/astropy-helpers/blob/9ba2f37481bdd2e5376531f91d261c6c35b9a2c5/astropy_helpers/openmp_helpers.py#L151-L156), where they are actually being used as flags passed to the compiler.

```python 
        # customize_compiler() extracts info from os.environ. If certain keys
        # exist it uses these plus those from sysconfig.get_config_vars().
        # If the key is missing in os.environ it is not extracted from 
        # sysconfig.get_config_var(). E.g. 'LDFLAGS' get left out, preventing
        # clang from finding libomp.dylib because -L<path> is not passed to linker.
        # Call get_openmp_flags() to get flags missed by customize_compiler().
```

Is this cool as is?